sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -954,6 +951,12 @@ class Analyzer(
     try {
       expr transformUp {
         case GetColumnByOrdinal(ordinal, _) => plan.output(ordinal)
+        case u @ UnresolvedAttribute(_) if resolver(u.name, VirtualColumn.hiveGroupingIdName) =>
```

Comment:
Could we do all the changes you made in this file in the rule `ResolveFunctions`?

Reply:
I don't think I can do it，because ResolveFunctions is behind ResolveGroupingAnalytics